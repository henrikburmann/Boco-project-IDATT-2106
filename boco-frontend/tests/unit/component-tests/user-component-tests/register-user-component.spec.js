import { mount } from "@vue/test-utils";
import RegisterFormComponent from "@/components/UserAuthComponents/RegisterForm";

describe("RegisterFormComponent", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(RegisterFormComponent);
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("is instantiated", () => {
    expect(wrapper.exists()).toBeTruthy();
  });

  it("renders the h2 text correctly", () => {
    expect(wrapper.find("#registerLabel").text()).toBe("Opprett ny konto");
  });

  it("has a button", () => {
    expect(wrapper.exists("button")).toBe(true);
  });

  it("updates data when field is updated", async () => {
    await wrapper.find('input[data-test="firstNameTest"]').setValue("Gunnar");
    expect(wrapper.vm.firstName).toBe("Gunnar");
  });

  /*   it("button click with correct sum", () => {
    wrapper.setData({ guess: "15" });
    const button = wrapper.find("button");
    button.trigger("click");
    expect(wrapper.vm.message).toBe("SUCCESS!");
  }); */
});
