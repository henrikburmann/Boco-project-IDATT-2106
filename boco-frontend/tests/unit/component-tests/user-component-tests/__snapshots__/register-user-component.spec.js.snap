// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterFormComponent renders correctly 1`] = `
<div
  data-v-app=""
>
  
  <!-- Register form for creating a new user account -->
  <div
    class="w-full max-w-md mx-auto mb-auto md:ring-1 ring-gray-300 overflow-hidden rounded-xl p-4"
  >
    <div
      class="text-xl md:text-2xl font-medium text-center text-primary-light mt-4 mb-8"
      id="registerLabel"
    >
       Opprett ny konto 
    </div>
    <form>
      <div
        class="grid grid-cols-1 gap-6 mt-4"
      >
        <div>
          <!-- Email -->
          <input
            class="block w-full px-4 py-2 mt-2 text-gray-700 placeholder-gray-500 bg-white border rounded-md dark:bg-gray-800 dark:border-gray-600 dark:placeholder-gray-400 focus:border-primary-light dark:focus:border-primary-light focus:ring-opacity-40 focus:outline-none focus:ring focus:ring-primary-light"
            id="email"
            placeholder="E-post adresse"
            type="email"
          />
          <!-- error message for email -->
          
          
          <div
            class="text-error-medium text-sm"
            style="display: none;"
          />
        </div>
        <div>
          <!-- Password -->
          <input
            class="block w-full px-4 py-2 mt-2 text-gray-700 placeholder-gray-500 bg-white border rounded-md dark:bg-gray-800 dark:border-gray-600 dark:placeholder-gray-400 focus:border-primary-light dark:focus:border-primary-light focus:ring-opacity-40 focus:outline-none focus:ring focus:ring-primary-light"
            id="password"
            placeholder="Passord"
            type="password"
          />
          <!-- error message -->
          
          
        </div>
        <div>
          <input
            class="block w-full px-4 py-2 mt-2 text-gray-700 placeholder-gray-500 bg-white border rounded-md dark:bg-gray-800 dark:border-gray-600 dark:placeholder-gray-400 focus:border-primary-light dark:focus:border-primary-light focus:ring-opacity-40 focus:outline-none focus:ring focus:ring-primary-light"
            id="confirmPassword"
            placeholder="Bekreft passord"
            type="password"
          />
          <!-- error message -->
          
          
        </div>
        <div>
          <!-- First name -->
          <input
            class="block w-full px-4 py-2 mt-2 text-gray-700 placeholder-gray-500 bg-white border rounded-md dark:bg-gray-800 dark:border-gray-600 dark:placeholder-gray-400 focus:border-primary-light dark:focus:border-primary-light focus:ring-opacity-40 focus:outline-none focus:ring focus:ring-primary-light"
            data-test="firstNameTest"
            id="firstName"
            placeholder="Fornavn"
            type="text"
          />
          <!-- error message for first name-->
          
          
        </div>
        <div>
          <!-- Last name -->
          <input
            class="block w-full px-4 py-2 mt-2 text-gray-700 placeholder-gray-500 bg-white border rounded-md dark:bg-gray-800 dark:border-gray-600 dark:placeholder-gray-400 focus:border-primary-light dark:focus:border-primary-light focus:ring-opacity-40 focus:outline-none focus:ring focus:ring-primary-light"
            id="lastName"
            placeholder="Etternavn"
            type="text"
          />
          <!-- error message for last name -->
          
          
        </div>
        <div>
          <!-- Address -->
          <input
            class="block w-full px-4 py-2 mt-2 text-gray-700 placeholder-gray-500 bg-white border rounded-md dark:bg-gray-800 dark:border-gray-600 dark:placeholder-gray-400 focus:border-primary-light dark:focus:border-primary-light focus:ring-opacity-40 focus:outline-none focus:ring focus:ring-primary-light"
            id="address"
            placeholder="Adresse"
            type="text"
          />
          <!-- error message for address -->
          
          
        </div>
      </div>
      <div
        class="flex justify-end mt-6"
      >
        
        <!-- Button with custom text and color -->
        <button
          class="flex items-center px-2 py-2 font-medium tracking-wide capitalize text-white transition-colors duration-200 transform rounded-md focus:outline-none focus:ring focus:ring-opacity-80 min-w-{20px} bg-primary-medium hover:bg-primary-dark focus:ring-primary-light"
        >
          Opprett
        </button>
        
      </div>
    </form>
  </div>
  
</div>
`;
